saleor/graphql/core/scalars.py

```diff
@@ -28,7 +28,7 @@ def parse_value(value):
             value = str(value)
             return decimal.Decimal(value)
         except decimal.DecimalException:
-            return None
+            return 0
```

Comment:
I'm not 100% about that. E.g. `ProductVariantInput` has `priceOverride` Decimal field and I think that we can pass `null` in API in order to "reset" the field. Setting `0` could override the price to zero. But that would have to be checked.

Reply:
That's how it has to be though. Decimal does not accept `None` or `""`. It will throw an exception.

Reply:
We need to make sure that this change won't introduce bugs in other parts of our codebase. I would double check how it works with `ProductVariantInput` and other occurrences of `Decimal` in API.

Reply:
I tested it with `ProductVariantInput`, if `null` is sent then the field is correctly nulled. If `""` then it sets it to 0 (before this change it would explode with `""`). 

In my opinion using `""` and expecting that the field is going to be nulled/zeroed is kinda too optimistic, but this change will at least ensure the API won't break. We should make sure that if we want to zero a field, then we send either `0` or `"0"`, not `""`. Ditto with `null`.

Reply:
So maybe we should raise an error if `""` is passed as Decimal? I think we could do that at the scalar implementation level.

Reply:
How is that different than not fixing this case? That's how it worked. If it was `""` it said it expected a Decimal and got an empty string.

```
    parse_result = type.parse_literal(value_ast)
    if parse_result is None:
        return [
            u'Expected type "{}", found {}.'.format(type.name, print_ast(value_ast))
        ]
```
from `graphql.utils` and then it reported an error:
`"message": "Variable \"$value\" got invalid value \"\".\nExpected type \"Decimal\", found \"\".",`

From the discussion we had I thought we want to support `""`, which resulted in this fix.

Sending `null` or `"0"` or `0` would alleviate this issue and assign `0` to the value (or `null` in case of a nullable field).

Reply:
I feel this is the same kind of a problem if one tried to assign a totally random number to a date field and expecting it to work.

Reply:
Right. I'm switching between different issues and I lost the context.